definitions/features/foreman_tasks.rb

```diff
@@ -22,8 +45,84 @@ def paused_tasks_count(ignored_tasks = [])
         WHERE state IN ('paused')
     SQL
     unless ignored_tasks.empty?
-      sql << "AND label NOT IN (#{ignored_tasks.map { |task| "'#{task}'" }.join(',')})"
+      sql << "AND label NOT IN (#{quotize(ignored_tasks)})"
     end
     feature(:foreman_database).query(sql).first['count'].to_i
   end
+
+  def count(state)
+    feature(:foreman_database).query(<<-SQL).first['count'].to_i
+     SELECT count(*) AS count FROM foreman_tasks_tasks WHERE #{condition(state)}
+    SQL
+  end
+
+  def delete(state)
+    tasks_condition = condition(state)
+
+    feature(:foreman_database).psql(<<-SQL)
+     BEGIN;
+       DELETE FROM dynflow_steps USING foreman_tasks_tasks WHERE (foreman_tasks_tasks.external_id = dynflow_steps.execution_plan_uuid) AND #{tasks_condition};
+       DELETE FROM dynflow_actions USING foreman_tasks_tasks WHERE (foreman_tasks_tasks.external_id = dynflow_actions.execution_plan_uuid) AND #{tasks_condition};
+       DELETE FROM dynflow_execution_plans USING foreman_tasks_tasks WHERE (foreman_tasks_tasks.external_id = dynflow_execution_plans.uuid) AND #{tasks_condition};
+       DELETE FROM foreman_tasks_tasks WHERE #{tasks_condition};
+     COMMIT;
+    SQL
+
+    count(state)
+  end
+
+  def condition(state)
+    raise 'Invalid State' unless valid(state)
+
+    if state == :old
+      old_tasks_condition
+    else
+      tasks_condition(state)
+    end
+  end
+
+  private
+
+  def backup_dir(state)
+    "/var/lib/foreman/backup-tasks/#{state}/#{Time.now.strftime('%Y-%m-%d_%H-%M-%S')}"
+  end
+
+  def backup_table(table, state, fkey = 'execution_plan_uuid')
+    yield("Backup #{table} [running]")
+    sql = "SELECT #{table}.* FROM foreman_tasks_tasks JOIN #{table} ON\
+       (foreman_tasks_tasks.external_id = #{table}.#{fkey})"
+    export_csv(sql, "#{table}.csv", state)
+    yield("Backup #{table} [DONE]")
+  end
+
+  def export_csv(sql, file_name, state)
+    dir = prepare_for_backup(state)
+    execute("echo \"COPY (#{sql}) TO STDOUT WITH CSV;\" \
```

Comment:
I would recommend rather going though `|` there, to avoid storing the original file (that already can be quite big) and saving just the compressed one.